### PR TITLE
add option to use rustls instead of native-tls in `shuttle-shared-db`

### PR DESCRIPTION
**Pull request by utterstep**
## Description of change

Adds feature `postgres-rustls` to `shuttle-shared-db`, allowing to use Postgres without relying on bindings to native SSL implementation.

Closes issue #690.

## How Has This Been Tested (if applicable)?

Created new example in https://github.com/shuttle-hq/shuttle-examples/pull/43, local run with both `postgres` and `postgres-rustls` feature is working:

`postgres`:
image

`postgres-rustls`:
image
image

### CODE REVIEW

**Comment by utterstep** [comment]
image
Now `test --all-feature` fails, as it includes mutually exclusive features..
Will think about possible workarounds

**Comment by utterstep** [comment]
Hmm, to be honest, I don't see an obvious way to solve this problem. This PR introduces feature `postgres-rustls` which is mutually exclusive with `postgres` bc the `sqlx` crate intentionaly fails to compile if both `rustls` and `native-tls` are present. 

And `cargo test --all-features` enables all features available.
Maybe consider splitting `shuttle-shared-db` tests in CI in two batches: with `native-tls` and with `rustls`?

@oddgrd what do you think about it?

**Comment by utterstep** [comment]
Split CI job for `shuttle-shared-db` into separate jobs for each feature.

Before (with mutually incompatible features from SQLx being selected by `--all-features`):
image

After:
image

Also it seems like `shuttle-deployer` tests are flaky?
Here are two consequent runs on my local machine without any diff between them:
image
image

File: .circleci/config.yml [hunk 1]
```diff
@@ -505,6 +508,16 @@ workflows:
                 - services/shuttle-tide
                 - services/shuttle-tower
                 - services/shuttle-warp
+      - check-standalone:
```
Comment (oddgrd) [inline .circleci/config.yml:511]: Maybe a short comment here about why we need this extra job could be useful?

**Review by oddgrd — APPROVED** [review]
Great, thanks for this @utterstep! :heart_eyes: Regarding the CI issue, I think your solution is good :ok_hand:

File: .circleci/config.yml (hunk 1 above)
Comment (utterstep) [reply .circleci/config.yml:511]: Sure, did that :)

**Pull request merged**